### DISCUSSION TITLE
How can I use `System.Windows.Media.Imaging` in WinUI3 apps?

### DISCUSSION THREAD

**Discussion by EFLKumo** [microsoft/WindowsAppSDK#2738] [Q&A] [answered]
Hi,

In **WPF** we can do sth like this:
```cs
using System.Windows.Media.Imaging;
...
var decoder = new IconBitmapDecoder(...);
...
```
When I put this code into a WinUI3 app, the error occurs. My IDE said that there wasn't namespace of `System.Windows.Media`, as the result, I can't use `IconBitmapDecoder` class in `System.Windows.Media.Imaging`.  
I've tried to use `Microsoft.UI.Xaml.Media.Imaging` instead, but it doesn't have classes like `IconBitmapDecoder` inside. I found that `System.Windows.Media.Imaging` is in `PresentationCore.dll` which may mean we can only use it in WPF programs. So is there any alternative for the code above?

**Comment by castorix** [comment, marked as answer]
It is [Windows.Graphics.Imaging.BitmapDecoder](https://docs.microsoft.com/en-us/uwp/api/windows.graphics.imaging.bitmapdecoder?view=winrt-22621)

**Reply by EFLKumo** [reply]
So it seems that I have to migrate the use of `IconBitmapDecoder` to `BitmapDecoder`. Thanks so much.
